yt_dlp/extractor/booyah.py

```diff
@@ -0,0 +1,89 @@
+from .common import InfoExtractor
+from ..utils import int_or_none, str_or_none, traverse_obj
+
+
+class BooyahBaseIE(InfoExtractor):
+    _BOOYAH_SESSION_KEY = None
+
+    def _real_initialize(self):
+        if not BooyahBaseIE._BOOYAH_SESSION_KEY:
+            initial_request = self._request_webpage(
+                'https://booyah.live/api/v3/auths/sessions', None, data="".encode())
+            BooyahBaseIE._BOOYAH_SESSION_KEY = initial_request.getheader('booyah-session-key')
+
+    def _get_comment_data(self, video_id):
+        comment_json = self._download_json(
+            f'https://booyah.live/api/v3/playbacks/{video_id}/comments/tops', video_id,
+            headers={'Booyah-Session-Key': BooyahBaseIE._BOOYAH_SESSION_KEY}, fatal=False) or []
+
+        return [{
+            'id': comment.get('comment_id'),
+            'author': comment.get('from_nickname'),
+            'author_id': comment.get('from_uid'),
+            'author_thumbnail': comment.get('from_thumbnail'),
+            'text': comment.get('content'),
+            'timestamp': comment.get('create_time'),
+            'like_count': comment.get('like_cnt'),
+        } for comment in comment_json.get('comment_list')] or None
```

Comment:
If `comment_json = []`, it will fail on `comment_json.get('comment_list')`

```suggestion
    def _get_comment_data(self, video_id):
        comment_json = self._download_json(
            f'https://booyah.live/api/v3/playbacks/{video_id}/comments/tops', video_id,
            headers={'Booyah-Session-Key': BooyahBaseIE._BOOYAH_SESSION_KEY}, fatal=False) or {}

        return [{
            'id': comment.get('comment_id'),
            'author': comment.get('from_nickname'),
            'author_id': comment.get('from_uid'),
            'author_thumbnail': comment.get('from_thumbnail'),
            'text': comment.get('content'),
            'timestamp': comment.get('create_time'),
            'like_count': comment.get('like_cnt'),
        } for comment in comment_json.get('comment_list') or ()]
```

Reply:
Didn't `for comment in comment_json.get('comment_list') or ()]` this give error ? as `()` this tuple right ?